assets/js/state/sagas/lastExecutions.js

```diff
@@ -34,7 +40,27 @@ export function* updateLastExecution({ payload }) {
 }
 
 export function* requestExecution({ payload }) {
-  yield put(setExecutionRequested(payload));
+  const { clusterID, navigate } = payload;
+  const clusterName = yield select(getClusterName(clusterID));
+
+  try {
+    yield call(triggerChecksExecution, clusterID);
+    yield put(setExecutionRequested(payload));
+    yield put(
+      notify({
+        text: `Checks execution requested, cluster: ${clusterName}`,
+        icon: '🐰',
+      })
+    );
+    navigate(`/clusters/${clusterID}/executions/last`);
```

Comment:
I don't really think we should do the `navigate` here.
I think we should use a promise chain:
```
dispatch(executionRequested(clusterID, hosts, selectedChecks))
.then(() => {
navigate(`/clusters/${clusterID}/executions/last`)
});
```

I know that this would increase some repeated code, but I think it would be the best.
Reducer functions should be as pure as possible.

Reply:
PD: I know that there was a conversation about this in the previous PR, but it still looks anti-pattern to me.

Reply:
Anyway, my comment comes most probably late as this was already discussed. Just take it as an external observer comment hehe

Reply:
I am not aware of the possibility to do `.then()` on a dispatch :eyes:, is that an option?

In total honesty, I am not a fan of this approach either (I came with a different solution https://github.com/trento-project/web/pull/1125), but still...

However I'd like this to not get stuck, so I am ok with this middleground.

Reply:
Quick look at our friend stackoverflow XD
https://stackoverflow.com/a/71582748

Reply:
I see, however I am afraid that's not our case cause if I try I get
```
Uncaught TypeError: dispatch(...).then is not a function
```
maybe we're missing something with `redux-toolkit`, which I would leave out from this PR.

Reply:
Leave it as it is @nelsonkopliku ,
I'm fine!

Reply:
@arbulu89 I wish this was so simple :smile: mainly here thing is that we are not using `redux-thunk`, a different middleware that allows dispatching asynchronous functions. (And even in that case, you can't `dispatch().then()` if I recall correctly, because `dispatch` isn't async itself)

When using Saga, navigating programmatically passing the `navigate` through the payload looks like something "socially accepted" in the ecosystem. The best thing would be having something like the `history` object available inside the sagas but it doesn't look like the middleware can be configured that way. Too bad.

If we discover something better, maybe we can pursue another path but for now let's just keep it this way.